test/micro/org/openjdk/bench/javax/crypto/full/Poly1305DigestBench.java

```diff
@@ -0,0 +1,95 @@
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package org.openjdk.bench.javax.crypto.full;
+
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Setup;
+
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.reflect.Method;
+import java.lang.reflect.Constructor;
+import java.security.Key;
+import java.security.spec.AlgorithmParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Warmup;
+
+@Warmup(time=3, iterations=5) // Reflection needs time to be upgraded to bytecodes
+@Fork(value = 1, jvmArgsAppend = {"--add-opens", "java.base/com.sun.crypto.provider=ALL-UNNAMED"})
+public class Poly1305DigestBench extends CryptoBase {
+    public static final int SET_SIZE = 128;
+
+    @Param({"64", "256", "1024", "" + 16*1024, "" + 1024*1024})
+    int dataSize;
+
+    private byte[][] data;
+    int index = 0;
+    private static MethodHandle polyEngineInit, polyEngineUpdate, polyEngineFinal;
+    private static Object polyObj;
+
+    static {
+        try {
+            MethodHandles.Lookup lookup = MethodHandles.lookup();
+            Class<?> polyClazz = Class.forName("com.sun.crypto.provider.Poly1305");
+            Constructor<?> constructor = polyClazz.getDeclaredConstructor();
+            constructor.setAccessible(true);
+            polyObj = constructor.newInstance();
+
+            Method m = polyClazz.getDeclaredMethod("engineInit", Key.class, AlgorithmParameterSpec.class);
+            m.setAccessible(true);
+            polyEngineInit = lookup.unreflect(m);
+
+            m = polyClazz.getDeclaredMethod("engineUpdate", byte[].class, int.class, int.class);
+            m.setAccessible(true);
+            polyEngineUpdate = lookup.unreflect(m);
+
+            m = polyClazz.getDeclaredMethod("engineDoFinal");
+            m.setAccessible(true);
+            polyEngineFinal = lookup.unreflect(m);
+        } catch (Throwable ex) {
+            throw new RuntimeException(ex);
+        }
+    }
+
+    @Setup
+    public void setup() {
+        setupProvider();
+        data = fillRandom(new byte[SET_SIZE][dataSize]);
+    }
+
+    @Benchmark
+    public byte[] digest() {
+        try {
+            byte[] d = data[index];
+            index = (index +1) % SET_SIZE;
+            polyEngineInit.invoke(polyObj, new SecretKeySpec(d, 0, 32, "Poly1305"), null);
+            polyEngineUpdate.invoke(polyObj, d, 0, d.length);
+            return (byte[])polyEngineFinal.invoke(polyObj);
+        } catch (Throwable ex) {
+            throw new RuntimeException(ex);
+        }
+    }
```

Comment:
On CLX patch shows performance regression of about 10% for block size 1024-2048+.

CLX (Non-IFMA target)

Baseline (JDK-20):-

Benchmark                   (dataSize)  (provider)   Mode  Cnt        Score   Error  Units
Poly1305DigestBench.digest          64              thrpt    2  3128928.978          ops/s
Poly1305DigestBench.digest         256              thrpt    2  1526452.083          ops/s
Poly1305DigestBench.digest        1024              thrpt    2   509267.401          ops/s
Poly1305DigestBench.digest        2048              thrpt    2   305784.922          ops/s
Poly1305DigestBench.digest        4096              thrpt    2   142175.885          ops/s
Poly1305DigestBench.digest        8192              thrpt    2    72142.906          ops/s
Poly1305DigestBench.digest       16384              thrpt    2    36357.000          ops/s
Poly1305DigestBench.digest     1048576              thrpt    2      676.142          ops/s


Withopt:
Benchmark                   (dataSize)  (provider)   Mode  Cnt        Score   Error  Units
Poly1305DigestBench.digest          64              thrpt    2  3136204.416          ops/s
Poly1305DigestBench.digest         256              thrpt    2  1683221.124          ops/s
Poly1305DigestBench.digest        1024              thrpt    2   457432.172          ops/s
Poly1305DigestBench.digest        2048              thrpt    2   277563.817          ops/s
Poly1305DigestBench.digest        4096              thrpt    2   149393.357          ops/s
Poly1305DigestBench.digest        8192              thrpt    2    79463.734          ops/s
Poly1305DigestBench.digest       16384              thrpt    2    41083.730          ops/s
Poly1305DigestBench.digest     1048576              thrpt    2      705.419          ops/s

Reply:
Odd, I measured it on `11th Gen Intel(R) Core(TM) i7-11700 @ 2.50GHz`, will go again